lib/utils.js

```diff
@@ -99,52 +102,35 @@ var getStringifiedStyles = function(rule) {
  */
 
 exports.getPreservedText = function(css, options) {
-  var rules = cssom.parse(css).cssRules || [];
-  var preserved = [];
-
-  for (var i = 0, l = rules.length; i < l; i++) {
-    /* CSS types
-      STYLE: 1,
-      IMPORT: 3,
-      MEDIA: 4,
-      FONT_FACE: 5,
-    */
-
-    if (options.fontFaces && rules[i].type === cssom.CSSFontFaceRule.prototype.type) {
-      var fontFace = [ '' ];
-      fontFace.push('@font-face {');
-      fontFace = fontFace.concat(getStringifiedStyles(rules[i]));
-      fontFace.push('}');
-
-      if (fontFace.length) {
-        preserved.push(fontFace.length ? fontFace.join(os.EOL) + os.EOL : '');
-      }
-    }
-
-    if (options.mediaQueries && rules[i].type === cssom.CSSMediaRule.prototype.type) {
-      var query = rules[i];
-      var queryString = [];
-
-      queryString.push(os.EOL + '@media ' + query.media[0] + ' {');
-
-      for (var ii = 0, ll = query.cssRules.length; ii < ll; ii++) {
-        var rule = query.cssRules[ii];
-
-        if (rule.type === cssom.CSSStyleRule.prototype.type
-            || rule.type === cssom.CSSFontFaceRule.prototype.type) {
-          queryString.push('  '
-            + (rule.type === cssom.CSSStyleRule.prototype.type ? rule.selectorText : '@font-face') + ' {');
-          queryString = queryString.concat(getStringifiedStyles(rule));
-          queryString.push('  }');
-        }
-      }
-
-      queryString.push('}');
-      preserved.push(queryString.length ? queryString.join(os.EOL) + os.EOL : '');
+  var parsed = mensch.parse(css, {position: true, comments: true});
+  var rules = typeof parsed.stylesheet != 'undefined' && parsed.stylesheet.rules ? parsed.stylesheet.rules : [];
+  var preserved = css;
+  var preserved2 = [];
+  var lastStart = null;
+
+  for (var i = rules.length - 1; i >= 0; i--) {
+    if (options.fontFaces && rules[i].type === 'font-face') {
+      // preserve
+      preserved2.push(mensch.stringify({ stylesheet: { rules: [ rules[i] ] }}, { comments: false, indentation: '    ' }).replace(/}/,os.EOL+'}'));
```

Comment:
Yes, it should be global.

Consider this is just to "mimic" the output from CSSOM and not something I would keep in a final release (the additional \n is just to make some test pass without alterning the ".out" file)

As you can see I compute the result in 2 different ways (one creates a new CSS from "keep rules", the other removes unwanted styles from the original style, so to keep formatting of the kept stuff "as is").

I don't know if you prefer to make a first "menschified" release that tries to mimic the old behaviour so we collect user feedback on a "minimal change" or if you instead prefer to kepp adding the changes and then make a final release including all of the major changes.